packages/entity/src/ComposedEntityCacheAdapter.ts

```diff
@@ -26,9 +26,13 @@ export default class ComposedEntityCacheAdapter<TFields> extends EntityCacheAdap
     fieldValues: readonly NonNullable<TFields[N]>[]
   ): Promise<ReadonlyMap<NonNullable<TFields[N]>, CacheLoadResult<TFields>>> {
     const retMap = new Map<NonNullable<TFields[N]>, CacheLoadResult<TFields>>();
+    const fulfilledFieldValuesByCacheIndex: NonNullable<TFields[N]>[][] = this.cacheAdapters.map(
+      () => []
+    );
 
     let unfulfilledFieldValues = fieldValues;
-    for (const cacheAdapter of this.cacheAdapters) {
+    for (let i = 0; i < this.cacheAdapters.length; i++) {
```

Comment:
We await within this loop to fetch less objects each time, though might it actually be faster to just fetch all from all caches in parallel as well and then merge them? Might be worth profiling. If we do stick with the for loop  we should probably have an early break when `unfulfilledFieldValues` is empty.

Reply:
Performance here will depend on the type of caches that are in the ComposedEntityCacheAdapter. In our case where we've got a local memory and redis cache, I would strongly prefer to keep the for-loop.  

If we fetched everything in parallel, we will be waiting for the slowest cache every single time (redis). In the case we have a popular query and the caches are warm, we'll be waiting on the redis cache even though all the values are available in local memory. This would be unfortunate because the redis cache is magnitudes slower than the local memory cache.

The for-loop will optimize for the case when the cache is warm (which will be the majority of requests for a popular query), since we will only be waiting on the local memory cache

Reply:
Good call. Yeah that makes sense.

We should still be able to get away with an early break out of the loop if `unfulfilledFieldValues` is empty though I think.